File: openacc/cuda_f/acckernel.f90
module acckernel
contains
  subroutine acckernel(N, C, A, B)
!    integer,value :: I, N
!    real(kind=4), device, dimension(N), intent(in) :: A, B
!    real(kind=4), device, dimension(N), intent(inout) :: C
    integer :: I, N
    real(4), device :: A(:), B(:), C(:)
!$acc kernels deviceptr(A,B,C)
!$acc loop
    do I=1, N
       C(I) = C(I) + A(I) * B(I)
    enddo
    !$acc end kernels
  end subroutine acckernel
end module acckernel

File: openacc/cuda_c/acckernel.c
void acckernel(int N, float *C, float *A, float *B)
{
#pragma acc kernels deviceptr(A,B,C)
#pragma acc loop independent
  for(int i=0; i<N; i++){
	C[i] += A[i] * B[i];
  }
}
